src/Moq/Obsolete/MockFactory.cs

```diff
@@ -384,13 +395,27 @@ protected virtual void VerifyMocks(Action<Mock> verifyAction)
 				}
 				catch (MockException error) when (error.Reason == MockExceptionReason.UnmatchedSetups)
 				{
-					errors.Add(error);
+					unmatchedSetups.Add(error);
+				}
+				catch (MockException error) when (error.Reason == MockExceptionReason.UnverifiedInvocations)
+				{
+					unverifiedInvocations.Add(error);
 				}
 			}
 
-			if (errors.Count > 0)
+			if (unmatchedSetups.Count > 0 && unverifiedInvocations.Count > 0)
+			{
+				throw MockException.MultipleVerificationErrors(unmatchedSetups.Concat(unverifiedInvocations));
```

Comment:
*If* we made `VerifyMocks` truly general, such that it could aggregate any number and type of verification error, we shouldn't even attempt to special-case `UnmatchedSetups` and `UnverifiedInvocations` (further below); we should just always do `MultipleVerificationErrors`. In that scenario, it would also be better to have only one `catch` clause (further above) and do a `when (error.IsVerificationError)`.

*However*, I don't think such a generic aggregation would pay off in the end, nor would it be a terribly clean as long as `MockException` nor the various verification error messages are designed with aggregation in mind. (Better to do that in a separate PR, as it would touch many parts of the code that aren't directly related to `VerifyNoOtherCalls`.)

I've always thought `VerifyMocks` was mis-designed, it's `protected virtual` but there's probably no point in ever overriding it. I think it would be better to just leave this method as is and accept that it's only really useful as an implementation detail of `Verify` and `VerifyAll`. That's why I suggested in another review comment (above) to just do `VerifyNoOtherCalls`-specific aggregation in `MockFactory.VerifyNoOtherCalls()` directly.